### PR TITLE
DLS-10832: Migrate to Dms Submission

### PR DESCRIPTION
**Pull request by johno1985**

### CODE REVIEW

**Comment by johno1985** [comment]
rebuild

**Comment by johno1985** [comment]
rebuild

**Comment by johno1985** [comment]
rebuild

**Comment by johno1985** [comment]
rebuild

**Comment by johno1985** [comment]
rebuild

**Review by ruthdas03 — CHANGES_REQUESTED** [review]

File: app/uk/gov/hmrc/cgtpropertydisposals/connectors/dms/DmsConnector.scala [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+      dataParts ++ fileParts ++ attachmentParts
+    )
+
+    logger.info("Sending payload to dms service...")
+    httpClient
+      .post(url"$url")
+      .setHeader(AUTHORIZATION -> servicesConfig.getString("internal-auth.token"))
+      .withBody(source)
+      .execute[DmsEnvelopeId]
+  }
```
[the remark this answers is not in the document]
Comment (johno1985) [reply app/uk/gov/hmrc/cgtpropertydisposals/connectors/dms/DmsConnector.scala:96]: We don't need to, as mentioned in the walkthrough yesterday, we don't need to handle anything, we just let the exception bubble to the top and let telemetry take over

File: app/uk/gov/hmrc/cgtpropertydisposals/config/InternalAuthTokenInitialiser.scala [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+              "resourceType"     -> "dms-submission",
+              "resourceLocation" -> "submit",
+              "actions"          -> List("WRITE")
+            )
+          )
+        )
+      )
+      .execute
+      .flatMap { response =>
+        if (response.status == 201) {
```
[the remark this answers is not in the document]
Comment (johno1985) [reply app/uk/gov/hmrc/cgtpropertydisposals/config/InternalAuthTokenInitialiser.scala:90]: Will do

File: app/uk/gov/hmrc/cgtpropertydisposals/controllers/dms/DmsCallbackController.scala [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+class DmsCallbackController @Inject() (cc: ControllerComponents) extends BackendController(cc) with Logging {
+  def callback: Action[DmsSubmissionResult] = Action.async(parse.json[DmsSubmissionResult]) { implicit request =>
+    val dmsSubmissionResult = request.body
+    dmsSubmissionResult.failureReason match {
+      case Some(failureReason) =>
+        logger.warn(
+          s"Dms submission failed due to $failureReason ${dmsSubmissionResult.failureType
+            .getOrElse("")}, id: ${dmsSubmissionResult.id}"
+        )
```
Comment (ruthdas03) [inline app/uk/gov/hmrc/cgtpropertydisposals/controllers/dms/DmsCallbackController.scala:35]: Do we need to add any monitoring for failures (point 11 in the jira ticket)

Comment (johno1985) [reply app/uk/gov/hmrc/cgtpropertydisposals/controllers/dms/DmsCallbackController.scala:35]: We will add a pager duty alert for this error message

**Review by ruthdas03 — APPROVED** [review]

**Pull request merged**